core/modules/link/link.module

```diff
@@ -813,30 +777,88 @@ function link_theme() {
 
 /**
  * Implements hook_element_info().
+ * 
+ * The #default_value should be an array with the following keys:
+ * @code
+ *   '#default_value' => array(
+ *     'url' => 'admin/content',
+ *     'title' => 'List of content',
+ *     'attributes' => array( // These are extra elements (see flags below)
+ *       'title' => 'A link to Content', // Prepopulates the "title" attribute.
+ *       'class' => 'additional-css-class',
+ *       'target' => '_blank',
+ *     ),
+ *   ),
+ * @endcode
  */
 function link_element_info() {
   $elements = array();
   // We use "link_field" here to avoid conflicts with theme_link().
+  // Many of these property are related to the Field API Link field settings.
   $elements['link_field'] = array(
     '#input' => TRUE,
     '#process' => array('link_field_process'),
     '#theme' => 'link_field',
     '#theme_wrappers' => array('form_element'),
+    '#element_validate' => array('link_field_element_validate'),
+
+    // Custom options for link_field. Defaults are provided within
+    // link_field_process().
+    '#link_field_options' => array(),
+
+    // #delta is typically passed by the widget form, but we provide a default
+    // value to help validations when link_field is used as a FAPI element.
+    '#delta' => 0,
+
+    // Attach the css by default.
+    '#attached' => array(
+      'css' => array(
+        backdrop_get_path('module', 'link') . '/css/link.css',
+      ),
+    ),
   );
   return $elements;
 }
 
 /**
- * Returns the default attributes and their values.
+ * Provides default values for #link_field_options properties.
  */
-function _link_default_attributes() {
+function _link_field_options_defaults() {
   return array(
+    // Possible values: 'optional' or 'required'. If 'optional', the element
+    // can be submitted without filling in the url textfield.
+    'url' => 'optional',
```

Comment:
Can this be made `'url_mode'`? Seems like that would bring it inline with `'title_mode'` below.